testhelpers/testhelpers.go

```diff
@@ -629,31 +629,22 @@ func SkipUnless(t *testing.T, expression bool, reason string) {
 func RunContainer(ctx context.Context, dockerCli client.CommonAPIClient, id string, stdout io.Writer, stderr io.Writer) error {
 	bodyChan, errChan := dockerCli.ContainerWait(ctx, id, container.WaitConditionNextExit)
 
-	if err := dockerCli.ContainerStart(ctx, id, dockertypes.ContainerStartOptions{}); err != nil {
-		return errors.Wrap(err, "container start")
-	}
-
-	info, err := dockerCli.Info(ctx)
+	logs, err := dockerCli.ContainerAttach(ctx, id, dockertypes.ContainerAttachOptions{
+		Stream: true,
+		Stdout: true,
+		Stderr: true,
+	})
 	if err != nil {
-		return errors.Wrap(err, "getting docker info")
-	}
-	if info.OSType == "windows" {
-		// wait for logs to show
-		time.Sleep(time.Second)
+		return err
 	}
 
-	logs, err := dockerCli.ContainerLogs(ctx, id, dockertypes.ContainerLogsOptions{
-		ShowStdout: true,
-		ShowStderr: true,
-		Follow:     true,
-	})
-	if err != nil {
-		return errors.Wrap(err, "container logs stdout")
+	if err := dockerCli.ContainerStart(ctx, id, dockertypes.ContainerStartOptions{}); err != nil {
+		return errors.Wrap(err, "container start")
 	}
 
 	copyErr := make(chan error)
 	go func() {
-		_, err := stdcopy.StdCopy(stdout, stderr, logs)
+		_, err := stdcopy.StdCopy(stdout, stderr, logs.Reader)
 		copyErr <- err
 	}()
```

Comment:
Use ContainerAttach + ContainerStart instead of ContainerStart + ContainerLogs. The current implementation has race conditions for Windows and is more heavy-weight for all OSes as the stdout/stderr have to flow through the logs subsystem. This implementation follows the docker/cli run [implementation](https://github.com/docker/cli/blob/f8696535e9e516a5a404661697fef6f1228ada64/cli/command/container/run.go#L157-L168) which effectively does the same (though with additional signal handling and tty options).

This fixes Windows, potentially avoids race conditions for both platforms by attaching before start, and I feel like having the Attach handle gives us more option for signalling that ContainerLogs handle in the future.